doc/ImplementationGuide_SmartNIC.md

```diff
@@ -0,0 +1,35 @@
+# Implementation Guide SmartNIC
+
+## Introduction
+This is a guide to implement a generic OpenOffload solution using the gRPC code provided in this repository. 
+
+### Assumptions
+This guide there is a virtual network function (VNF) connected to a SmartNIC. The VNF is responsible for maintaining its own flow table of all sessions, and depending on internal polic, which is invisible to the SmartNIC instructing the SmartNIC to perform actions on the corresponding flows. The initial actions defined in the  gRPC module are:
+
+- DROP: Drop all packets in this session.
+- FORWARD: Forward (allow) all packets in this session to go to final destination.
+- MIRROR: Create a copy of the packet and send to a traffic collector.
+- SNOOP:
+
+## Sequence of Operations
+
+The following is the sequence of operations for OpenOffload.
+
+### Register
+The VNF implements the server side of the Activatation service. It is assumed that in the future a VNF may have more than one OpenOffload device registered. 
+
+The goal of the registration process is to provide the VNF with information to securely connect to the OpenOffload device and to provide the VNF administrator with some metadata about the vendor, device type and capabilities. No checking is done on the metadata it is only used as a guide to the admin. The information about how ot securely connect to the device is used in the activation phase. 
+
+### Activate
+From the available OpenOffload devices registered the VNF administrator can select one to be activated. For version 1 the number of devices that can be activated by a VNF is limited to one, this may be changed in future versions.
+
+Once the device has been activated it uses the information supplied in the registration process to securely connect to the OpenOffload device.
+
+### Add Session
+Once the device is activated the VNF can decide to offload flows to the OpenOffload device. To do this the VNF sends an addSession message to the activsated OpenOffload device. At this point the VNF should not see any payload packets from the OpenOffload device, all packets for the session should be handled by the offload device.
```

Comment:
but there may be packets in flight that still need to be processed by the applciation until the device takes over the session flow.